Source/ACE.Server/WorldObjects/Player_Use.cs

```diff
@@ -18,18 +24,23 @@ public void HandleActionUseWithTarget(uint sourceObjectGuid, uint targetObjectGu
         {
             StopExistingMoveToChains();
 
-            var invSource = GetInventoryItem(sourceObjectGuid);
-            var invTarget = GetInventoryItem(targetObjectGuid);
-            var invWielded = GetEquippedItem(targetObjectGuid);
+            // source item is always in our possession
+            var sourceItem = FindObject(sourceObjectGuid, SearchLocations.MyInventory | SearchLocations.MyEquippedItems, out _, out _, out var sourceItemIsEquipped);
 
-            if (invSource == null)
+            if (sourceItem == null)
             {
-                // is this caster with a built-in spell?
-                var caster = GetEquippedItem(sourceObjectGuid);
```

Comment:
This should be caster / GetEquippedItem here, else players could send this packet to cast spells from items they don't have equipped